lib/LANraragi/Model/Tankoubon.pm

```diff
@@ -283,6 +223,50 @@ sub delete_tankoubon ($tank_id) {
     }
 }
 
+# update_tankoubon(name, data)
+#   Updates metadata and archive list.
+#   Returns 1 on success, 0 on failure alongside an error message.
+sub update_tankoubon ( $tank_id, $data ) {
+
+    my ( $result, $err ) = update_metadata( $tank_id, $data );
+    if ($result) {
+        my ( $result, $err ) = update_archive_list( $tank_id, $data );
+    }
+
+    return ( $result, $err);
+}
+
+# update_metadata(tankoubonid, data)
+#   Updates the metadata in the Tankoubon.
+#   Returns 1 on success, 0 on failure alongside an error message.
+sub update_metadata( $tank_id, $data ) {
+
+    my $logger          = get_logger( "Tankoubon", "lanraragi" );
+    my $redis           = LANraragi::Model::Config->get_redis;
+    my $err             = "";
+    my $name            = $data->{"metadata"}->{"name"};
```

Comment:
I didn't catch it the first time but requiring all the fields to be set is pretty annoying for API clients and doesn't match how the update archive endpoint functions -- Would it be possible to change that?  

Feels like we could just undef check before calling update_meta for each field

Reply:
Yeah, I did it this way because the rule is that for PUT operations, you should set all the fields, while PATCH ones are the ones you send only one field, but I guess it's not necessary to follow that here. I'll do the change.